### PR TITLE
[pt2] add metas for `grid_sampler_3d` ops

### PR DESCRIPTION
**Pull request by nkaretnikov**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #106261

### CODE REVIEW

File: torch/_meta_registrations.py [hunk 1]
```diff
@@ -4031,6 +4032,70 @@ def meta_max_pool3d_with_indices_backward(
...
+        input.ndim == 5 and input.ndim == grid.ndim,
+        lambda: (
+            f"grid_sampler(): expected 5D input and grid with same number of "
+            f"dimensions, but got input with sizes {input.shape}",
+            f" and grid with sizes {grid.shape}",
+        ),
+    )
+    torch._check(
+        not (
+            input.dim() == 5
```
Comment (nkaretnikov) [inline torch/_meta_registrations.py:4092]: nit: use ndim

File: torch/_meta_registrations.py [hunk 2]
```diff
@@ -4031,6 +4032,70 @@ def meta_max_pool3d_with_indices_backward(
...
+    BICUBIC = 2
+
+
+def check_grid_sampler_3d(input: Tensor, grid: Tensor, interpolation_mode: int):
+    torch._check(
+        input.ndim == 5 and input.ndim == grid.ndim,
+        lambda: (
+            f"grid_sampler(): expected 5D input and grid with same number of "
+            f"dimensions, but got input with sizes {input.shape}",
+            f" and grid with sizes {grid.shape}",
```
Comment (nkaretnikov) [inline torch/_meta_registrations.py:4087]: will fix: extra commas (tuple)

**Comment by nkaretnikov** [comment]
From the list in https://github.com/pytorch/pytorch/issues/105105

File: torch/_meta_registrations.py [hunk 3]
```diff
@@ -4031,6 +4032,70 @@ def meta_max_pool3d_with_indices_backward(
...
+        lambda: (
+            f"grid_sampler(): expected input and grid to have torch.strided layout, but "
+            f"input has {input.layout} and grid has {grid.layout}"
+        ),
+    )
+    torch._check(
+        input.shape[0] == grid.shape[0],
+        lambda: (
+            "grid_sampler(): expected grid and input to have same batch size, but got "
+            "input with sizes {input.shape} and grid with sizes {grid.shape}"
```
Comment (nkaretnikov) [inline torch/_meta_registrations.py:4054]: Will fix: needs f-strings

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**